### PR TITLE
Add Hiarcs Chess Explorer (v1.8e)

### PR DESCRIPTION
**Pull request by webpro**
Commercial chess software.

### CODE REVIEW

**Comment by radeksimko** [comment]
Well done on your first contribution!

Just a few things to be sorted out before we can merge this:
 - [`uninstall` Is Required for Casks That Install a pkg](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#uninstall-is-required-for-casks-that-install-a-pkg) otherwise the cask cannot be uninstalled cleanly
 - the app for unknown reason doesn't react to `QUIT` signal and keeps running, so I'd suggest [using `TERM`](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#uninstall-key-signal)
 - is there any reason why do you use `(Deep)` in the `name` stanza?

Hint:

```ruby
  uninstall :signal => ['TERM', 'com.hiarcs.chessexplorer'],
            :pkgutil => 'com.hiarcs.*'
```

**Comment by webpro** [comment]
Thanks for your feedback.

* Tested and added the `uninstall` stanza.
* Tested and added the `TERM` exit signal mechanism.
* The "(Deep)" is there, since the same download file/application is used to install either of the [commercial product versions](http://www.hiarcs.com/mac-chess-explorer.htm). So while I agree it's not looking great, people can see this cask is to be used for both.

Want me to squash the commits?

**Comment by radeksimko** [comment]
> Want me to squash the commits?

Yes, please, then it's good to go.

**Comment by webpro** [comment]
Allright, squashed.

File: Casks/hiarcs-chess-explorer.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cask :v1 => 'hiarcs-chess-explorer' do
+  version '1.8e'
+  sha256 '53dc3f7db266c1400dc2d5aab3a05eb8c503a50b4cdbc18e57d3d6d9ae0ce129'
+
+  url "http://www.hiarcs.com/mhce/HIARCS-Chess-Explorer-Installer-v#{version}.pkg"
+  pkg "HIARCS-Chess-Explorer-Installer-v#{version}.pkg"
```
Comment (vitorgalvao) [inline Casks/hiarcs-chess-explorer.rb]: This should be before `uninstall`, as in

```ruby
cask :v1 => 'hiarcs-chess-explorer' do
  version '1.8e'
  sha256 '53dc3f7db266c1400dc2d5aab3a05eb8c503a50b4cdbc18e57d3d6d9ae0ce129'

  url "http://www.hiarcs.com/mhce/HIARCS-Chess-Explorer-Installer-v#{version}.pkg"
  name '(Deep) HIARCS Chess Explorer'
  homepage 'http://www.hiarcs.com/mac-chess-explorer.htm'
  license :commercial

  pkg "HIARCS-Chess-Explorer-Installer-v#{version}.pkg"

  uninstall :signal => ['TERM', 'com.hiarcs.chessexplorer'],
            :pkgutil => 'com.hiarcs.*'
end
```

**Comment by webpro** [comment]
OK, moved `pkg`.

**Comment by vitorgalvao** [comment]
Thank you. Everything looks great.

**Pull request merged**